lib/scoring.rb

```diff
@@ -1,9 +1,81 @@
+
+LETTER_VALUES = {
+  "A"=>1, "B"=>3, "C"=>3, "D"=>2,
+  "E"=>1, "F"=>4, "G"=>2, "H"=>4,
+  "I"=>1, "J"=>8, "K"=>5, "L"=>1,
+  "M"=>3, "N"=>1, "O"=>1, "P"=>3,
+  "Q"=>10, "R"=>1, "S"=>1, "T"=>1,
+  "U"=>1, "V"=>4, "W"=>4, "X"=>8,
+  "Y"=>4, "Z"=>10
+}
+
 module Scrabble
   class Scoring
+
     def self.score(word)
+
+      word = word.upcase
+      letters = word.split("")
+      character_check = LETTER_VALUES.keys
+      values_of_letters = []
+
+      letters.each do |character|
+
+        # Checking if character input is a valid letter or not
+        # First part is splitting the hash into an array of the
+        # key values 'A-Z' and then we are checking if any characters
+        # are outside that range (This includes blank or nothing).
+        # If so, automatically return nil
+        #
+        # Else, if all the characters are valid letters, take the value
+        # associated with the key and put that in an array of integers
+
+
+        if !character_check.include?(character) || letters.length == 0 || letters.length > 7
```

Comment:
This `if` doesn't need to be in the loop, at least the checks for the length of the array.